kedro-datasets/kedro_datasets/xarray/geotiff_dataset.py

```diff
@@ -0,0 +1,133 @@
+"""GeoTiff loads and saves data to a local geoiff file. The
+underlying functionality is supported by rioxarray and xarray. A read geotiff file
+returns a xarray.DataArray object.
+"""
+
+from copy import deepcopy
+from pathlib import PurePosixPath
+from typing import Any
+
+import fsspec
+import rioxarray as rxr
+import xarray
+from kedro.io import AbstractVersionedDataset, DatasetError
+from kedro.io.core import Version, get_filepath_str, get_protocol_and_path
+
+
+class GeoTiffDataset(AbstractVersionedDataset[xarray.DataArray, xarray.DataArray]):
```

Comment:
Wondering if this should be `GeoTIFFDataset` to follow conventional naming for GeoTIFF. For example, our NetCDF PR was `NetCDFDataset`.